File: src/api/analytics.ts
import { request } from './client';

export async function trackEvent(eventType: string, properties?: Record<string, any>, sessionId?: string) {
  try {
    const res = await request('/analytics/track', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ event_type: eventType, properties, session_id: sessionId }),
    });
    return res.json();
  } catch (_) { return { success: false }; }
}

export async function getAnalyticsDaily(date: string) {
  const res = await request(`/analytics/daily/${date}`);
  return res.json();
}

export async function getAnalyticsRange(from: string, to: string) {
  const res = await request(`/analytics/range?from=${from}&to=${to}`);
  return res.json();
}

export async function getAnalyticsSummary(days = 30) {
  const res = await request(`/analytics/summary?days=${days}`);
  return res.json();
}

export async function getAnalyticsEventCounts(days = 30) {
  const res = await request(`/analytics/event-counts?days=${days}`);
  return res.json();
}

export async function getAnalyticsRecentEvents(limit = 50) {
  const res = await request(`/analytics/recent-events?limit=${limit}`);
  return res.json();
}

export async function getAnalyticsDashboard() {
  const res = await request('/analytics/dashboard');
  return res.json();
}

File: src-tauri/src/bridge/routes/analytics.rs
use axum::{routing::{get, post}, Router};
use super::super::AppState;
pub fn routes() -> Router<AppState> {
    Router::new()
        .route("/api/analytics/dashboard", get(super::super::analytics_dashboard))
        .route("/api/analytics/track", post(super::super::analytics_track_event))
        .route("/api/analytics/daily/{date}", get(super::super::analytics_daily))
        .route("/api/analytics/range", get(super::super::analytics_range))
        .route("/api/analytics/summary", get(super::super::analytics_summary))
        .route("/api/analytics/event-counts", get(super::super::analytics_event_counts))
        .route("/api/analytics/recent-events", get(super::super::analytics_recent_events))
}
